drools-wb-screens/drools-wb-appformer-extensions-editor/drools-wb-appformer-extensions-editor-backend/src/main/java/org/drools/workbench/screens/appformer/extensions/backend/PerspectiveExtensionServiceImpl.java

```diff
@@ -0,0 +1,223 @@
+import org.uberfire.ext.editor.commons.service.RenameService;
+import org.uberfire.ext.layout.editor.api.LayoutServices;
+import org.uberfire.ext.layout.editor.api.editor.LayoutTemplate;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.file.FileAlreadyExistsException;
+import org.uberfire.rpc.SessionInfo;
+import org.uberfire.workbench.events.ResourceOpenedEvent;
+
+@Service
+@ApplicationScoped
+public class PerspectiveExtensionServiceImpl
+        extends KieService<PerspectiveExtensionContent>
+        implements PerspectiveExtensionService {
+
+    @Inject
+    @Named("ioStrategy")
+    private IOService ioService;
+
+    @Inject
+    private CopyService copyService;
+
+    @Inject
+    private DeleteService deleteService;
+
+    @Inject
+    private RenameService renameService;
+
+    @Inject
+    private Event<ResourceOpenedEvent> resourceOpenedEvent;
+
+    @Inject
+    private GenericValidator genericValidator;
+
+    @Inject
+    private CommentedOptionFactory commentedOptionFactory;
+
+    private SafeSessionInfo safeSessionInfo;
+
+    @Inject
+    private LayoutServices layoutServices;
+
+    public PerspectiveExtensionServiceImpl() {
+    }
+
+    @Inject
+    public PerspectiveExtensionServiceImpl(final SessionInfo sessionInfo) {
+        safeSessionInfo = new SafeSessionInfo(sessionInfo);
+    }
+
+    @Override
+    public Path create(final Path context,
+                       final String fileName,
+                       final String content,
+                       final String comment) {
+        try {
+
+            final org.uberfire.java.nio.file.Path nioPath = Paths.convert(context).resolve(fileName);
+            final Path newPath = Paths.convert(nioPath);
+
+            if (ioService.exists(nioPath)) {
+                throw new FileAlreadyExistsException(nioPath.toString());
+            }
+
+            ioService.write(nioPath,
+                            content,
+                            commentedOptionFactory.makeCommentedOption(comment));
+
+            return newPath;
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public String load(final Path path) {
+        try {
+            final String content = ioService.readAllString(Paths.convert(path));
+
+            return content;
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public PerspectiveExtensionContent loadContent(final Path path) {
+        return super.loadContent(path);
+    }
+
+    @Override
+    protected PerspectiveExtensionContent constructContent(Path path,
+                                                           Overview overview) {
+
+        String content = load(path);
+        LayoutTemplate layoutTemplate = layoutServices.convertLayoutFromString(content);
+
+        resourceOpenedEvent.fire(new ResourceOpenedEvent(path,
```

Comment:
We should probably convince ourselves these events are still used. They were when we have an "Inbox" feature; however that was removed in 7.0.0.Final. I am less certain however whether "Social Activities" still use these events. If they're not used by "Social" please could you create a GUVNOR JIRA saying we can remove use of "Resource Opened Events" (and assign to me if you like).